quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImpl.java

```diff
@@ -30,4 +32,42 @@ public Collection addCollection(String name, String description) {
         return collectionRepo.save(collection);
     }
 
+    /**
+     * Finds a collection from mongodb collection table by _id
+     * 
+     * @param id used to look up the collection in mongodb
+     * 
+     * 
+     * @return Collection Instance that is found from the database
+     * 
+     **/
+    @Override
+    public Collection findCollection(String id) {
+        return collectionRepo.findBy_id(new ObjectId(id));
+    }
+
+    /**
+     * Edits an existing Collection and updates it in mongodb
+     * 
+     * @param _id of the collection that needs to be updated
+     * 
+     * 
+     * @return Collection Instance that is updated in database
+     * 
+     **/
+    @Override
+    public Collection editCollection(String id, Optional<String> name, Optional<String> description) {
+        Optional<Collection> collection = Optional.ofNullable(findCollection(id));
+        Collection updatedCollection = null;
+
+        if (collection.isPresent()) {
+            Collection currentCollection = collection.get();
+            name.ifPresent(updatedName -> currentCollection.setName(updatedName));
+            description.ifPresent(updatedDescription -> currentCollection.setDescription(updatedDescription));
+            updatedCollection = collectionRepo.save(currentCollection);
```

Comment:
just return here and return null at the end of the method. In fact, I would throw an exception if a collection with the provided id doesn't exist.